pallets/providers/src/utils.rs

```diff
@@ -1196,21 +1413,22 @@ impl<T: pallet::Config> MutateBucketsInterface for pallet::Pallet<T> {
     }
 
     fn remove_root_bucket(bucket_id: Self::BucketId) -> DispatchResult {
-        let bucket = Buckets::<T>::take(&bucket_id).ok_or(Error::<T>::BucketNotFound)?;
+        let bucket = Buckets::<T>::get(&bucket_id).ok_or(Error::<T>::BucketNotFound)?;
 
-        MainStorageProviderIdsToBuckets::<T>::mutate_exists(
-            &bucket.msp_id,
-            |buckets| match buckets {
-                Some(b) => {
-                    b.retain(|b| b != &bucket_id);
+        let msp_id = match bucket.msp_id {
+            Some(msp_id) => msp_id,
+            None => return Err(Error::<T>::BucketMustHaveMspForOperation.into()),
+        };
```

Comment:
I agree, not sure what was the reason behind this.

Reply:
Modified it so that it is not required but if an msp is present, the fixed rate payment stream is updated.

```rust
if let Some(msp_id) = bucket.msp_id {
            Self::apply_delta_fixed_rate_payment_stream(
                &msp_id,
                &bucket_id,
                &bucket.user_id,
                RateDeltaParam::RemoveBucket,
            )?;

            MainStorageProviderIdsToBuckets::<T>::remove(msp_id, &bucket_id);
        };
```